sqldelight-core/src/main/kotlin/com/squareup/sqldelight/core/lang/util/QueryUtil.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.core.lang.util
+
+import com.alecstrong.sqlite.psi.core.psi.NamedElement
+import com.alecstrong.sqlite.psi.core.psi.SqliteExpr
+import com.alecstrong.sqlite.psi.core.psi.SqliteQueryElement.QueryResult
+import com.intellij.psi.PsiElement
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier.ABSTRACT
+import java.util.LinkedHashSet
+
+/**
+ * Explodes a sqlite query into an ordered list (same order as the query) of abstract methods with
+ * names and types corresponding to the query.
+ */
+internal fun List<QueryResult>.flatFunctions(): List<FunSpec> {
+  val namesUsed = LinkedHashSet<String>()
+
+  return flatMap {
```

Comment:
This block is unnecessary.

Reply:
is there something simpler that replaces it? One `QueryResult` -> `List<FunSpec>` so I'm flatmapping to go from `List<QueryResult>` -> `List<FunSpec>`

Reply:
Sorry thought this was a named block. LGTM.